src/lib.rs

```diff
@@ -874,104 +1165,153 @@ impl<T> Grid<T> {
     }
 
     /// Converts self into a vector without clones or allocation.
+    ///
+    /// The order of the elements depends on the internal memory layout, which is
+    /// row-major by default.
     #[must_use]
     pub fn into_vec(self) -> Vec<T> {
         self.data
     }
 
     /// Transpose the grid so that columns become rows in new grid.
-    #[must_use]
-    pub fn transpose(&self) -> Grid<T>
-    where
-        T: Clone,
-    {
-        let mut data = Vec::with_capacity(self.data.len());
-        for c in 0..self.cols {
-            for r in 0..self.rows {
-                data.push(self[r][c].clone());
+    ///
+    /// This method changes the internal memory layout.
+    pub fn transpose(&mut self) {
```

Comment:
Had no benchmark for this func before. I would guess it is a few magnitues faster now :-)